src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivitiesResourceIntTest.kt

```diff
@@ -161,6 +176,97 @@ class ActivitiesResourceIntTest : IntegrationTestBase() {
     }
   }
 
+  @Nested
+  inner class UpdateActivity {
+
+    private fun updateActivityRequest() = UpdateActivityRequest(prisonId = "LEI")
+
+    @Test
+    fun `access forbidden when no authority`() {
+      webTestClient.put().uri("/activities/1")
+        .contentType(MediaType.APPLICATION_JSON)
+        .body(BodyInserters.fromValue(updateActivityRequest()))
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `access forbidden when no role`() {
+      webTestClient.put().uri("/activities/1")
+        .contentType(MediaType.APPLICATION_JSON)
+        .headers(setAuthorisation(roles = listOf()))
+        .body(BodyInserters.fromValue(updateActivityRequest()))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `access forbidden with wrong role`() {
+      webTestClient.put().uri("/activities/1")
+        .contentType(MediaType.APPLICATION_JSON)
+        .headers(setAuthorisation(roles = listOf("ROLE_BANANAS")))
+        .body(BodyInserters.fromValue(updateActivityRequest()))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `should call the service`() {
+      doReturn(UpdateActivityResponse(prisonId = "LEI")).whenever(activityService).updateActivity(anyLong(), any())
```

Comment:
would it better to create the data for the activity as part of the test rather than using a mock/spy bean?  Just thinking about the convention of the other integration tests.